homeassistant/components/wiser/__init__.py

```diff
@@ -0,0 +1,362 @@
+                        vol.Coerce(int)
+                    ),
+                    vol.Optional(CONF_BOOST_TEMP, default=DEFAULT_BOOST_TEMP): vol.All(
+                        vol.Coerce(int)
+                    ),
+                    vol.Optional(
+                        CONF_BOOST_TEMP_TIME, default=DEFAULT_BOOST_TEMP_TIME
+                    ): vol.All(vol.Coerce(int)),
+                }
+            ],
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """
+    Wiser uses config flow for configuration.
+
+    But, a "wiser:" entry in configuration.yaml will trigger an import flow
+    if a config entry doesn't already exist. If it exists, the import
+    flow will attempt to import it and create a config entry, to assist users
+    migrating from the old wiser component. Otherwise, the user will have to
+    continue setting up the integration via the config flow.
+    """
+    hass.data.setdefault(DOMAIN, {})
+
+    if DOMAIN not in config:
+        return True
+
+    for conf in config[DOMAIN]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf,
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up Wiser from a config entry."""
+    _async_import_options_from_data_if_missing(hass, config_entry)
+
+    data = WiserHubHandle(hass, config_entry,)
+
+    try:
+        await hass.async_add_executor_job(data.connect)
+    except KeyError:
+        _LOGGER.error("Failed to login to wiser hub")
+        return False
+    except RuntimeError as exc:
+        _LOGGER.error("Failed to setup wiser hub: %s", exc)
+        return ConfigEntryNotReady
+    except requests.exceptions.HTTPError as ex:
+        if ex.response.status_code > 400 and ex.response.status_code < 500:
+            _LOGGER.error("Failed to login to wiser hub: %s", ex)
+            return False
+        raise ConfigEntryNotReady
+
+    # Do first update
+    await hass.async_add_executor_job(data.update)
+
+    # Poll for updates in the background
+    update_track = async_track_time_interval(
+        hass,
+        lambda now: data.update(),
+        timedelta(
+            seconds=config_entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+        ),
+    )
+
+    update_listener = config_entry.add_update_listener(_async_update_listener)
+
+    hass.data[DOMAIN][config_entry.entry_id] = {
+        DATA: data,
+        UPDATE_TRACK: update_track,
+        UPDATE_LISTENER: update_listener,
+    }
+
+    for platform in WISER_PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, platform)
+        )
+
+    _LOGGER.info("Wiser Component Setup Completed")
+    await data.async_update_device_registry()
+
+    return True
+
+
+@callback
+def _async_import_options_from_data_if_missing(hass, config_entry):
```

Comment:
This should be part of an options flow?

Reply:
amended config flow setup and options and removed this